tools/distrib/generate_boringssl_prefix_header.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+# Copyright 2018 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Generate the list of boringssl symbols that need to be renamed based on the
+# current boringssl submodule. The script should be run after a boringssl
+# upgrade in third_party/boringssl-with-bazel. Note that after the script is
+# run, you will typically need to manually upgrade the BoringSSL-GRPC podspec
+# (templates/src/objective-c/BoringSSL-GRPC.podspec.template) version and the
+# corresponding version number in gRPC-Core podspec
+# (templates/gRPC-Core.podspec.template).
+
+set -ev
+
+BORINGSSL_ROOT=third_party/boringssl-with-bazel/src
+
+cd "$(dirname $0)"
+cd ../../$BORINGSSL_ROOT
+
+BORINGSSL_COMMIT=$(git rev-parse HEAD)
+BORINGSSL_PREFIX_HEADERS_DIR=src/boringssl
+
+# generate the prefix header
+rm -rf build
+mkdir -p build
+cd build
+cmake ..
+make -j
+
+[ -f ssl/libssl.a ] || { echo "Failed to build libssl.a" ; exit 1 ; }
```

Comment:
please add a note where this magic is coming from. Without more context, this seems very arbitrary and hacky.

Reply:
What magic do you mean? Do you mean the path where `libssl.a` is in?

Reply:
sorry for not being clearer. by "magic" I meant the sequence of commands (build, call read_symbols.go, build again, ...) that someone without context will have no idea where they came from. If you just mention you got the commands from the boringssl docs for prefixing symbols then all is fine I think.